src/gluonts/model/rotbaum/_predictor.py

```diff
@@ -0,0 +1,184 @@
+
+# Relative imports
+from ._preprocess import PreprocessOnlyLagFeatures
+from ._model import QRX
+
+
+class RotbaumForecast(Forecast):
+    """
+    Implements the quantile function in Forecast for TreePredictor,
+    as well as a new estimate_dists function for estimating a sampling of the
+    conditional distribution of the value of each of the steps in the
+    forecast horizon (independently).
+    """
+
+    @validated()
+    def __init__(
+        self,
+        models: List,
+        featurized_data: List,
+        start_date: pd.Timestamp,
+        freq,
+        prediction_length: int,
+    ):
+        self.models = models
+        self.featurized_data = featurized_data
+        self.start_date = start_date
+        self.freq = freq
+        self.prediction_length = prediction_length
+        self.item_id = None
+        self.lead_time = None
+
+    def quantile(self, q: float) -> np.array:
+        """
+        Returns np.array, where the i^th entry is the estimate of the q
+        quantile of the conditional distribution of the value of the i^th
+        step in the forecast horizon.
+        """
+        assert 0 <= q <= 1
+        return np.array(
+            list(
+                chain.from_iterable(
+                    model.predict(self.featurized_data, q)
+                    for model in self.models
+                )
+            )
+        )
+
+    def estimate_dists(self) -> np.array:
+        """
+        Returns np.array, where the i^th entry is an estimated sampling from
+        the conditional distribution of the value of the i^th step in the
+        forecast horizon.
+        """
+        return np.array(
+            list(
+                chain.from_iterable(
+                        model.estimate_dist(self.featurized_data)
+                        for model in self.models
+                )
+            )
+        )
+
+
+class TreePredictor(RepresentablePredictor):
+    """
+    A predictor that uses a QRX model for each of the steps in the forecast
+    horizon. (In other words, there's a total of prediction_length many
+    models being trained. In particular, this predictor does not learn a
+    multivariate distribution.) The list of these models is saved under
+    self.model_list.
+    """
+
+    @validated()
+    def __init__(
+        self,
+        context_length: Optional[int],
+        prediction_length: Optional[int],
+        n_ignore_last: int = 0,
+        lead_time: int = 0,
+        max_n_datapts: int = 1000000,
+        clump_size: int = 100,
+        max_workers=None,
+        model_params=None,
+        freq=None,
+    ) -> None:
+        self.lead_time = lead_time
+        self.preprocess_object = PreprocessOnlyLagFeatures(
+            context_length,
+            forecast_horizon=prediction_length,
+            stratify_targets=False,
+            n_ignore_last=n_ignore_last,
+            max_n_datapts=max_n_datapts,
+        )
+        self.context_length = context_length
```

Comment:
You can add these common asserts from the other models.
```
assert (
            context_length is None or context_length > 0
        ), "The value of `context_length` should be > 0"
        assert (
            prediction_length > 0
        ), "The value of `prediction_length` should be > 0"
```

Reply:
and also 
```
self.context_length = (
            context_length
            if context_length is not None
            else self.prediction_length
        )
```
as an example or which the best default for `context_length` is.

Reply:
Done.